File: src/main/java/org/ecews/renaissance/domain/Optometrist.java
/*
* MIT License
*
* Copyright (c) 2025 [AUTHOR OR ORGANIZATION]
*
* Permission is hereby granted, free of charge, to any person obtaining a copy
* of this software and associated documentation files (the "Software"), to deal
* in the Software without restriction, including without limitation the rights
* to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
* copies of the Software, and to permit persons to whom the Software is
* furnished to do so, subject to the following conditions:
*
* The above copyright notice and this permission notice shall be included in
* all copies or substantial portions of the Software.
*
* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
* IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
* FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
* AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
* LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
* OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
* THE SOFTWARE.
*/
package org.ecews.renaissance.domain;

import jakarta.persistence.*;
import lombok.Data;
import org.hibernate.annotations.GenericGenerator;

import java.util.UUID;

@Data
@Entity
@Table(name = "ren_optometrist")
public class Optometrist extends AuditEntity {
    
    @Id
    @GeneratedValue
    @GenericGenerator(name = "uuid2", strategy = "uuid2")
    @Column(columnDefinition = "BINARY(16)")
    private UUID id;
    
    @ManyToOne(optional = false)
    @JoinColumn(name = "patient_id")
    private Patient patient;
    
    @Column(name = "visual_acuity_right")
    private String visualAcuityRight;
    
    @Column(name = "visual_acuity_left")
    private String visualAcuityLeft;
    
    @Column(name = "glasses_dispensed")
    private Boolean glassesDispensed;
    
    @Column(name = "referred")
    private Boolean referred;
}

File: Renaissance.AspNet/backend/Renaissance.Domain/Entities/Optometrist.cs
using System.Text.Json.Serialization;

namespace Renaissance.Domain.Entities;

public class Optometrist : AuditEntity
{
    public Guid Id { get; set; }
    public Guid PatientId { get; set; }

    [JsonIgnore]
    public Patient? Patient { get; set; }

    public string? VisualAcuityRight { get; set; }
    public string? VisualAcuityLeft { get; set; }
    public List<string> Diagnoses { get; set; } = [];
    public List<string> Treatments { get; set; } = [];
    public List<string> Services { get; set; } = [];
    public List<string> Medications { get; set; } = [];
    public bool? GlassesDispensed { get; set; }
    public bool? Referred { get; set; }
}
